File: Selenium/Activities/Activity11.py
import time

from selenium import webdriver
from selenium.webdriver.common.by import By
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Open the browser to the URL
    driver.get("https://v1.training-support.net/selenium/dynamic-controls")
    print("Title of the page is: ", driver.title)
    # Find the checkbox and verify if selected
    checkbox = driver.find_element(By.NAME, "toggled")
    print("Is checkbox is selected: ", checkbox.is_selected())
    # click the checkbox
    checkbox.click()
    time.sleep(2)
    print("Is checkbox is selected: ", checkbox.is_selected())

File: Selenium/Activities/Activity11.java
package Activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity11 {
    public static void main(String[] args) throws InterruptedException {
        // Setup the Firefox driver(GeckoDriver)
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/dynamic-controls");
        System.out.println("Title of the page: " + driver.getTitle());
        // Find the checkbox input element
        WebElement checkbox = driver.findElement(By.name("toggled"));
        System.out.println("Is checkbox is selected: " +checkbox.isSelected());
        // Click the checkbox
        checkbox.click();
        Thread.sleep(1000);
        System.out.println("Is checkbox is selected: " +checkbox.isSelected());
        driver.quit();
    }
}
